src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateSupportedRuntime.targets

```diff
@@ -0,0 +1,63 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.GenerateSupportedRuntime.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="WriteAppConfigWithSupportedRuntime" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <UsingTask TaskName="SetGeneratedAppConfigMetadata" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+
+  <PropertyGroup>
+    <_GenerateSupportedRuntimeIntermediateAppConfig>$(IntermediateOutputPath)$(TargetFileName).withSupportedRuntime.config</_GenerateSupportedRuntimeIntermediateAppConfig>
+  </PropertyGroup>
+
+  <Target Name="GenerateSupportedRuntime"
+          Condition="'$(GenerateSupportedRuntime)' != 'false' and '$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(HasRuntimeOutput)' == 'true'"
+          DependsOnTargets="_WriteAppConfigWithSupportedRuntime"
+          BeforeTargets="GenerateBindingRedirects">
```

Comment:
@rainersigwald this is the issue I asked. We are worried about if this target can still run when GenerateBindingRedirects is disable.

Reply:
> Before a target is executed or skipped, any target that lists it in a BeforeTargets attribute is run.

https://docs.microsoft.com/en-us/visualstudio/msbuild/target-build-order#determine-the-target-build-order

@nguerrera plus I tested out disable AutoGenerateBindingRedirects=false . it is fine to use BeforeTargets="GenerateBindingRedirects"

Reply:
Ah, ok, I always get confused by that.